p2pserver/block_sync.go

```diff
@@ -345,17 +464,27 @@ func (this *BlockSyncMgr) OnHeaderReceive(headers []*types.Header) {
 	err := this.ledger.AddHeaders(headers)
 	this.delFlightHeader(height)
 	if err != nil {
+		this.addErrorRespCnt(fromID)
 		log.Errorf("OnHeaderReceive AddHeaders error:%s", err)
 		return
 	}
 	this.syncHeader()
 }
 
 //OnBlockReceive receive block from net
-func (this *BlockSyncMgr) OnBlockReceive(block *types.Block) {
+func (this *BlockSyncMgr) OnBlockReceive(fromID uint64, block *types.Block) {
 	height := block.Header.Height
 	blockHash := block.Hash()
 	log.Debugf("OnBlockReceive Height:%d", height)
+	this.recvBlkNodes[blockHash] = fromID
+	flightInfos := this.flightBlocks[blockHash]
+	for _, flightInfo := range flightInfos {
+		if flightInfo.GetNodeId() == fromID {
+			t := flightInfo.GetStartTime()
+			go this.addNewSpeed(fromID, block, t)
```

Comment:
Because addNewSpeed need to serialize block and it is a expensive operation. And the network speed is decided by the block size and the time spent, it seems the serialization is unavoidable。

Reply:
modify `message.go:ReadMessage` function to return `(Message, PayloadSize, error)`, the PayloadSize is just `hdr.Length`,  and add a `PayloadSize` field in `types.MsgPayload` struct.

Reply:
the code has updated, thanks